### PR TITLE
feat(fan profile): #83 is ready for test;

### PR DESCRIPTION
**Pull request by DiLiUA**

### CODE REVIEW

File: src/fan-profile/fan-profile.component.ts [hunk 1]
```diff
@@ -1,36 +1,119 @@
 import { Component, OnInit } from '@angular/core';
-import { Subscription } from 'rxjs/Subscription';
-import { FanProfileService } from './fan-profile.service';
+
+import * as moment from 'moment';
+
```
Comment (Ugmaxie) [inline src/fan-profile/fan-profile.component.ts]: please sort imports

File: src/fan-profile/fan-profile.component.ts [hunk 2]
```diff
@@ -1,36 +1,119 @@
...
 
-  public constructor(fanProfileService: FanProfileService) {
-    this.fanProfileService = fanProfileService;
   }
 
-  public ngOnInit(): void {
-    this.radioModel = 'permonth';
+  ngOnInit() {
+    this.route.queryParams
+      .subscribe(params => {
```
Comment (Ugmaxie) [inline src/fan-profile/fan-profile.component.ts:44]: if you can use filter here - use it

File: src/fan-profile/fan-profile.component.ts [hunk 3]
```diff
@@ -1,36 +1,119 @@
...
+    this.joinDate = this.user.joinDate
+      ? moment(this.user.joinDate).format('dddd, MMMM DD YYYY')
+      : undefined;
+
+    this.dateOfBirth = this.user.dateOfBirth
+      ? moment(this.user.dateOfBirth).format('dddd, MMMM DD YYYY')
+      : undefined;
+  }
+
+  getBackedShows(): void | undefined {
```
Comment (Ugmaxie) [inline src/fan-profile/fan-profile.component.ts]: void | undefined => void

return undefined => return void

File: src/fan-profile/fan-profile.component.ts [hunk 4]
```diff
@@ -1,36 +1,119 @@
...
+    const backedShowsSubscription = this.showService.getEventsListByQuery(query);
+    backedShowsSubscription
+      .subscribe(res => {
+        this.backedShows = res.map(show => this.getShowInfo(show));
+      }, err => {
+        console.error('something went wrong: ', err);
+      });
+  }
+
+  getAttendedShows(): void | undefined {
```
Comment (Ugmaxie) [inline src/fan-profile/fan-profile.component.ts]: void. undefined... as above

File: src/shared/show-info/show-info.component.ts [hunk 5]
```diff
@@ -7,11 +7,15 @@ import { ShowInfoDate } from './showInfoDate.interface';
 import { Config } from '../../app.config';
 import { Show } from '../show-service/show.model';
 import { User } from '../../user-service/user.model';
-import { Statistics } from '../statistics/statistics.interface';
-import { LocalStorageService } from '../index';
+import { StatisticsItem } from '../statistics/statistics.interface';
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.ts]: please sort imports

File: src/shared/statistics/statistics.component.ts [hunk 6]
```diff
@@ -1,46 +1,39 @@
 import { Component, Input, OnInit } from '@angular/core';
-import { LikeRequestObj, Statistics } from './statistics.interface';
+import { LikeRequestObj, StatisticsItem } from './statistics.interface';
```
Comment (Ugmaxie) [inline src/shared/statistics/statistics.component.ts]: please sort imports

File: src/shared/statistics/statistics.component.ts [hunk 7]
```diff
@@ -1,46 +1,39 @@
...
-        this.user = null;
-
-        return undefined;
-      }
-
-      this.user = JSON.parse(userProfile);
-    } catch (err) {
-      console.log('err: ', err);
+  setLike(item: StatisticsItem): void | undefined {
+    if (item.title !== 'Likes') {
```
Comment (Ugmaxie) [inline src/shared/statistics/statistics.component.ts]: void, undefined...

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/fan-profile/fan-profile.component.ts (hunk 2 above)
Comment (DiLiUA) [reply src/fan-profile/fan-profile.component.ts:44]: this is router subscribe,  I think use  filter for this subscribe doesn't have a sense

File: src/fan-profile/fan-profile.component.html [hunk 8]
```diff
@@ -1,145 +1,107 @@
...
+      <div class="right-part">
+        <div class="user-statistics-container">
+          <app-statistics [statistics]="userStatistics" [id]="user._id" [isEvent]="false" [isTitle]="true"></app-statistics>
         </div>
 
-        <div class="get-link">
-          <img src="/assets/img/get-link.png" alt="">
-          <div class="shares-number">Get link</div>
+        <div class="social-links-container"
+             *ngIf="!!user.socials.facebook || !!user.socials.twitter || !!user.socials.google || !!user.socials.instagram">
```
Comment (Ugmaxie) [inline src/fan-profile/fan-profile.component.html]: What if user has facebook but doesn't have any other socials?

There's an idea to check'em separately.

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**